### PR TITLE
Feature/us8266 leader can end participant

### PR DESCRIPTION
**Pull request by JoeKer1**

### CODE REVIEW

**Comment by vchtelmakh** [comment]
"Feature/us8266 leader can end participant"

Added appropriate label for "end participant"

File: src/app/components/pin-details/gathering/participant-details/participant-details.component.html [hunk 1]
```diff
@@ -1,22 +1,31 @@
-<div class="connect-container connect-layout-container is-gathering">
-<div class="row pin-details-dark dark-theme">
-    <div class="col-xs-12 soft media media-dark">
-        <profile-picture imageClass="media-object img-size-6" wrapperClass="media-img-wrapper" [contactId]="participant.contactId"></profile-picture>
-        <div class="media-body">
-            <h2 class="media-heading media-heading-stacked">
+<div class="connect-container connect-layout-container is-gathering" *ngIf="ready">
```
Comment (vchtelmakh) [inline src/app/components/pin-details/gathering/participant-details/participant-details.component.html]: Should be 'readyEh' or 'isReady', may also want to specify what it is that is ready.

File: src/app/components/pin-details/gathering/participant-details/participant-details.component.ts [hunk 2]
```diff
@@ -13,52 +13,60 @@ import { Component, Input, OnInit } from '@angular/core';
...
                         this.state.setPageHeader('Participant', this.redirectUrl);
                         this.state.setLoading(false);
-                    })
-                    .subscribe(
-                        (address) => {
-                            this.participantAddress = address;
-                        }, error => {
-                            console.log(error);
-                        });
+                    }).subscribe(address => {
```
Comment (vchtelmakh) [inline src/app/components/pin-details/gathering/participant-details/participant-details.component.ts]: You have a nested observable here, imo, the `next` event (upon receiving successful reply from getPartialPersonAddress) observable should be in a function. It's hard to immediately follow what's happening. This *is* nitpicky and you don't need to change it.

File: src/app/components/pin-details/gathering/participant-remove/participant-remove.component.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            this.participantService.getGroupParticipant(this.groupId, this.groupParticipantId).finally(() => {
+                this.state.setLoading(false);
+            })
+            .subscribe(p => {}, error => {
+                console.log(error);
+                this.handleError();
+            });
+            this.redirectUrl = `/${this.router.url.split('/')[1]}/${this.groupId}`;
+        }, error => {
+            this.handleError();
```
Comment (vchtelmakh) [inline src/app/components/pin-details/gathering/participant-remove/participant-remove.component.ts]: Should loading be set to false here?

File: src/app/components/pin-details/gathering/participant-details/participant-details.component.html (hunk 1 above)
Comment (JoeKer1) [inline src/app/components/pin-details/gathering/participant-details/participant-details.component.html]: OH yeah I forgot about that, putting that in there gets rid of console errors. probably a better way to do it but it works for now

File: src/app/components/pin-details/gathering/participant-remove/participant-remove.component.ts (hunk 3 above)
Comment (JoeKer1) [inline src/app/components/pin-details/gathering/participant-remove/participant-remove.component.ts]: it's in the finally block above

File: src/app/components/pin-details/gathering/participant-details/participant-details.component.ts (hunk 2 above)
Comment (JoeKer1) [inline src/app/components/pin-details/gathering/participant-details/participant-details.component.ts]: Lets get together on Monday just so I'm sure of what you're saying :D

**Pull request merged**